### PR TITLE
Ccw0

### PR DESCRIPTION
**Pull request by mducoffe**
Hi,

I added a state _epoch_interrupt_received to the status. Let me know if everything is okay for this file.

Mélanie

### CODE REVIEW

File: blocks/main_loop.py [hunk 1]
```diff
@@ -246,13 +247,13 @@ def _check_finish_training(self, level):
...
         if (self.log.current_row.training_finish_requested or
-                self.log.current_row.batch_interrupt_received or
-                self.log.previous_row.batch_interrupt_received):
-            raise TrainingFinish
-        if (level == 'epoch' and
-                (self.log.current_row.epoch_interrupt_received or
-                 self.log.previous_row.epoch_interrupt_received)):
-            raise TrainingFinish
+	     self.log.current_row.batch_interrupt_received or
+	     self.log.previous_row.batch_interrupt_received):
```
Comment (rizar) [inline blocks/main_loop.py]: Can you please also introduce a `_batch_interrupt_received` record in the status, similarly to `_epoch_interrupt_received`? This looking into the previous row is not really nice, my bad :)

Comment (mducoffe) [inline blocks/main_loop.py]: I will do it this afternoon, thanks for your fast answer =)

File: blocks/main_loop.py [hunk 2]
```diff
@@ -260,6 +261,9 @@ def _handle_epoch_interrupt(self, signal_number, frame):
                        epoch_interrupt_message)
         signal.signal(signal.SIGINT, self._handle_batch_interrupt)
         self.log.current_row.epoch_interrupt_received = True
+	# add epoch_interrupt_received to the status so to keep track on it after training
```
Comment (rizar) [inline blocks/main_loop.py]: I would rephrase as 

```Add a record to the statu. Unlike the log record it will be easy to access at later iterations.```

**Comment by rizar** [comment]
It looks nice! Please pay attention the results of code analysis, to see them simply click on the  `Scrutiner` button above, scroll down and see a few things that need to be fixed.

File: blocks/main_loop.py [hunk 3]
```diff
@@ -246,13 +248,11 @@ def _check_finish_training(self, level):
         # the iteration the corresponding log record can be found only in
         # the previous row.
         if (self.log.current_row.training_finish_requested or
-                self.log.current_row.batch_interrupt_received or
-                self.log.previous_row.batch_interrupt_received):
-            raise TrainingFinish
+            self.status._batch_interrupt_received):
```
Comment (bartvm) [inline blocks/main_loop.py]: I see you ran into a nasty part of PEP8. This should work:
```python
if (self.log.current_row.training_finish_requested or
        self.status._batch_interrupt_received):
    raise TrainingFinish
```
The indentation should be allowed like this in order to avoid `self.status` starting at the same point as `raise`.

**Comment by rizar** [comment]
@mducoffe , I would recommend you to use the `flake8` tool to check syntax errors locally. First install it with `pip` (`pip install --user flake 8`) and then run it in the folder where you cloned repository is. Saves a lot of time compared to getting syntax errors from Scrutinizer.

**Comment by rizar** [comment]
@bartvm , do you know what does this message from Travis log mean?

```
tests.test_examples.FunctionTestCase (test_mnist) ... /home/travis/build.sh: line 41:  2829 Segmentation fault      (core dumped) coverage run -p --source=blocks -m nose2.__main__ -v tests
```

**Comment by mducoffe** [comment]
@rizar, I usually used pyflakes but don't know why it wasn't detected the errors of Scrutinizer. I will use flake8 instead

**Comment by bartvm** [comment]
@rizar Sorry, that was my fault :) the `__getattr__` function instead of the properties on `IndexableDataset` actually caused an infinite recursion in a few very specific cases. It's fixed now.

@mducoffe `flake8` runs both pyflakes and pep8; pyflakes tests for logical errors (imports not being used, variables not defined) while pep8 checks for formatting errors (whitespace, line breaks)

**Comment by bartvm** [comment]
Tests pass, and I just tested it and it seems to work well. Merging!

**Comment by mducoffe** [comment]
[?] !!!!!

2015-02-27 13:30 GMT-05:00 Bart van Merriënboer <notifications@github.com>:

**Comment by bartvm** [comment]
Why all the exclamation marks? :p

**Comment by mducoffe** [comment]
Because it is my first pull request ever

2015-02-27 14:12 GMT-05:00 Bart van Merriënboer <notifications@github.com>:
